test/test_overall_state.py

```diff
@@ -0,0 +1,592 @@
+                       headers=headers_patch, auth=self.auth)
+
+        # Get service
+        response = requests.get(self.endpoint + '/service', params=sort_id, auth=self.auth)
+        resp = response.json()
+        r = resp['_items']
+        ls_service = copy.copy(r[0])
+
+        # ---------
+        # Update live state for a service
+        # => OK SOFT (anyway, do not care about HARD state!)
+        time.sleep(0.1)
+        data = {
+            'ls_state': 'OK',
+            'ls_state_id': 0,
+            'ls_state_type': 'SOFT',
+            'ls_last_check': 1465685852,
+            'ls_last_state': 'OK',
+            'ls_last_state_type': 'HARD',
+            'ls_output': 'CRITICAL - Plugin timed out after 10 seconds',
+            'ls_long_output': '',
+            'ls_perf_data': '',
+            'ls_acknowledged': False,
+            'ls_execution_time': 10.0598139763,
+            'ls_latency': 1.3571469784
+        }
+        headers_patch = {
+            'Content-Type': 'application/json',
+            'If-Match': ls_service['_etag']
+        }
+        requests.patch(self.endpoint + '/service/' + ls_service['_id'], json=data,
+                       headers=headers_patch, auth=self.auth)
+        response = requests.get(
+            self.endpoint + '/service/' + ls_service['_id'], params=sort_id, auth=self.auth
+        )
+        ls_service = response.json()
+        # _overall_state_id field is 0
+        self.assertEqual(0, ls_service['_overall_state_id'])
+
+        # Get host overall state
+        response = requests.get(
+            self.endpoint + '/host/' + ls_host['_id'], params=sort_id, auth=self.auth
+        )
+        ls_host = response.json()
+        # _overall_state_id field is 0
+        self.assertEqual(0, ls_host['_overall_state_id'])
+
+        # => WARNING SOFT (anyway, do not care about HARD state!)
+        time.sleep(0.1)
+        data = {
+            'ls_state': 'WARNING',
+            'ls_state_id': 1,
+            'ls_state_type': 'SOFT',
+            'ls_last_check': 1465685852,
+            'ls_last_state': 'OK',
+            'ls_last_state_type': 'HARD',
+            'ls_output': 'CRITICAL - Plugin timed out after 10 seconds',
+            'ls_long_output': '',
+            'ls_perf_data': '',
+            'ls_acknowledged': False,
+            'ls_execution_time': 10.0598139763,
+            'ls_latency': 1.3571469784
+        }
+        headers_patch = {
+            'Content-Type': 'application/json',
+            'If-Match': ls_service['_etag']
+        }
+        requests.patch(self.endpoint + '/service/' + ls_service['_id'], json=data,
+                       headers=headers_patch, auth=self.auth)
+        response = requests.get(
+            self.endpoint + '/service/' + ls_service['_id'], params=sort_id, auth=self.auth
+        )
+        ls_service = response.json()
+        # _overall_state_id field is 3
+        self.assertEqual(3, ls_service['_overall_state_id'])
```

Comment:
It's WARNING SOFT, not sure we must put it `_overall_state_id` in `3`, the question is about manage it only with HARD?